### PR TITLE
allocator-recipe-resolver supports mapping manifest stores

### PR DESCRIPTION
**Pull request by mmandlis**
addresses b/165637376

and:
- remove support for copy fate stores.
- small type var resolution fix (follow up to #5940)

### CODE REVIEW

File: src/tools/allocator-recipe-resolver.ts [hunk 1]
```diff
@@ -93,14 +93,22 @@ export class AllocatorRecipeResolver {
...
     }
+    for (const store of this.runtime.context.allStores) {
+      if (handleById[store.id]) {
+        throw new AllocatorRecipeResolverError(
+            `Handle with ${store.id} already exists`);
+      }
+      handleById[store.id] = {store, handles: []};
+    }
+
     // Find all `map` and `copy` handles and add them to the map.
```
Comment (alxrsngrtn) [inline src/tools/allocator-recipe-resolver.ts:104]: Does this comment need to be updated?

Comment (mmandlis) [reply src/tools/allocator-recipe-resolver.ts]: yes, done, thanks.

**Comment by mmandlis** [comment]
@shans FYI I'm merging now, if you have any comments, I'll address them in follow up PRs.

File: src/tools/allocator-recipe-resolver.ts [hunk 2]
```diff
@@ -112,17 +120,19 @@ export class AllocatorRecipeResolver {
     // Iterate over all handles in the recipe set that are shared across
     // multiple recipes, and compute their restricted type and apply them to
     // all relevant handles and connections.
-    for (const value of Object.values(handleById)) {
+    for (const handleId of Object.keys(handleById)) {
       // tslint:disable-next-line: no-any
-      const {createHandle, handles} = value as any;
+      const {createHandle, store, handles} = handleById[handleId] as any;
```
Comment (shans) [inline src/tools/allocator-recipe-resolver.ts:125]: `as {Handle, AbstractStore, Handle[]}`?

File: src/tools/allocator-recipe-resolver.ts [hunk 3]
```diff
@@ -112,17 +120,19 @@ export class AllocatorRecipeResolver {
...
       const restrictedType =
-          this.restrictHandleType(createHandle.id, [createHandle, ...handles]);
+          this.restrictHandleType(handleId, [createHandle, store, ...handles].filter(h => !!h));
```
Comment (shans) [inline src/tools/allocator-recipe-resolver.ts:127]: why can some of these be null?

File: src/tools/allocator-recipe-resolver.ts (hunk 2 above)
Comment (shans) [reply src/tools/allocator-recipe-resolver.ts:125]: or, are you trying to turn store into a {type, direction} pair? If so you can probably type accordingly. It's good to use the types though, they pick up bugs.

**Comment by piotrswigon** [comment]
What's the context of this PR? I presume here we just handle the mapping of an explicit store, but will the store actually be created in the StorageService with the data from the `json` file? I haven't heard of anyone doing the necessary work.

**Comment by mmandlis** [comment]
> What's the context of this PR? I presume here we just handle the mapping of an explicit store, but will the store actually be created in the StorageService with the data from the `json` file? I haven't heard of anyone doing the necessary work.

@alxrsngrtn was trying to use a mapped store, Alex, could you elaborate what/how were you trying to do?

**Comment by alxrsngrtn** [comment]
> > What's the context of this PR? I presume here we just handle the mapping of an explicit store, but will the store actually be created in the StorageService with the data from the `json` file? I haven't heard of anyone doing the necessary work.
> 
> @alxrsngrtn was trying to use a mapped store, Alex, could you elaborate what/how were you trying to do?

Sure. I was primarily focused on how to structure `arcs_kt_gen` / `arcs_kt_plan` rules & dependencies. I wanted to test if it was possible to import a manifest that had a store. Also, would it be possible to load a manifest that loads a `.json` file? How would our rules recognize the runtime dependency. 

The manifest in question was trying to load a constant from a json file for use within a recipe. I set a minimum value, and I have particles that filter for entities that have a field above the minimum value.

**Pull request merged**